newrelic/resource_newrelic_synthetics_private_location.go

```diff
@@ -0,0 +1,193 @@
+package newrelic
+
+import (
+	"context"
+	"log"
+
+	"github.com/newrelic/newrelic-client-go/pkg/common"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/synthetics"
+)
+
+func resourceNewRelicSyntheticsPrivateLocation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicSyntheticsPrivateLocationCreate,
+		ReadContext:   resourceNewRelicSyntheticsPrivateLocationRead,
+		UpdateContext: resourceNewRelicSyntheticsPrivateLocationUpdate,
+		DeleteContext: resourceNewRelicSyntheticsPrivateLocationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The ID of the account in New Relic.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The private location description.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the private location.",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"verified_script_execution": {
+				Type:        schema.TypeBool,
+				Required:    true,
```

Comment:
I'm thinking we move this to optional and add a default as 'false'. Most customers won't need or use this.

What do you think @mbazhlekova and @sanderblue?

Reply:
Agreed